A43-sdk-authorization.md

```diff
@@ -827,77 +615,101 @@ public final class FileAuthorizationServerInterceptor implements ServerIntercept
+        .addService(service)
+        .intercept(authzServerInterceptor)
+        .build()
+        .start();
 ```
 
-5. RBAC Engine
+- Policy is reloaded dynamically from file path
 
 ```Java
-package io.grpc.xds.internal.rbac;
-
-public class AuthorizationDecision {
-  public enum Decision {
-    ALLOW,
-    DENY,
-  }
-  private Decision decision;
-  private String matchingPolicyName;
-}
-
-public class RbacEngine {
-  public RbacEngine(RBAC policy);
-  // Given a RPC call and headers, evaluate the policy.
-  public AuthorizationDecision evaluate(
-      ServerCall<ReqT, RespT> call, Metadata headers) {
-  }
+ScheduledExecutorService scheduledExecutor =
+    Executors.newSingleThreadScheduledExecutor(
+        new ThreadFactoryBuilder()
+            .setNameFormat(...)
+            .setDaemon(true)
+            .build());
+FileAuthorizationServerInterceptor authzServerInterceptor;
+try {
+  authzServerInterceptor = FileAuthorizationServerInterceptor.create(authzPolicyFile);
+} catch (IllegalArgumentException iae) {
+  // ...
+} catch (IOException ioe) {
+  // ...
 }
+Closeable closeable = authzServerInterceptor.scheduleRefreshes(
+    1, TimeUnit.HOURS, scheduledExecutor);
+Server server =
+    Grpc.newServerBuilderForPort(port, serverCreds)
+        .addService(service)
+        .intercept(authzServerInterceptor)
+        .build()
+        .start();
+// In the end, free up the resources used for policy refresh.
+closeable.close();
 ```
 
 ## Rationale
 
+gRPC Authorization internally implements RBAC Engine(s) based on Envoy [RBAC policy](https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/rbac/v3/rbac.proto).
 We decided to create a new policy language "SDK authorization policy" instead of 
-consuming Envoy RBAC directly due to following reasons
+consuming Envoy RBAC directly due to following reasons:
 - Envoy RBAC is a complex language, and we preferred using a simple human readable 
   policy language
 - With our own language, we can provide a stable API, even when Envoy undergoes 
   versioning updates.
 
-Authorization APIs take policy in JSON format, instead of protobuf. gRPC wants to
-avoid the dependency on protobuf when possible. We have users in OSS that use gRPC
-without protobuf. Another reason is to have a consistent API across languages.
+Note that SDK authorization policy is a subset of Envoy RBAC, and it does not
+support all the fields that are present in Envoy RBAC. 
+
+RBAC policy provides service-level and method-level access control for a service.
+Engines process incoming RPC request attributes against policy configs and make a
+decision on whether to allow or deny the request. The decision depends on the type
+of policy (if the policy is an allowlist or denylist) and whether a matching policy
+was found.
+
+This engine implementation is shared by SDK-based authorization and xDS-based
+authorization. In gRPC Proxyless Service Mesh (or xDS authorization), Istio or
```

Comment:
I don't think it makes sense for this document to reference things like "Proxyless Service Mesh" or TD or Istio.  Instead, it should stick to discussing open-source protocols like xDS.  So this can say something like, "The xDS server sends an RBAC policy config to the xDS-enabled gRPC server for authorization."

Reply:
Sure. Updated text. Thank you!